azure/hub/variables.tf

```
@@ -169,7 +169,7 @@ variable "virtual_network_gateway_config" {
 
   validation {
     error_message = "If the sku is set to Basic, VpnGw1 or VpnGw1AZ, then generation must be set to Generation1."
-    condition     = (var.virtual_network_gateway_config == null || contains(["Basic", "VpnGw1", "VpnGw1AZ"], var.virtual_network_gateway_config.sku) && var.virtual_network_gateway_config.generation == "Generation1")
+    condition     = (var.virtual_network_gateway_config == null || can(contains(["Basic", "VpnGw1", "VpnGw1AZ"], var.virtual_network_gateway_config.sku) && var.virtual_network_gateway_config.generation == "Generation1"))
```

If I'm reading this correctly, `["Basic", "VpnGw1", "VpnGw1AZ"]` are the only valid values?

If this is the case, the error message isn't clear enough. It should be not conditional.

---

Also, I'm slightly confused by the choice of brackets in the original format.

It' was written as:
`(A OR B AND C)`

but does it mean:

`(A OR B) AND C`

or

`A OR (B AND C)`

The `can` function clears this up somewhat so maybe the brackets aren't needed anymore?

I think moving the brackets around might make the expression more human friendly.

---

That said however, if we're taking the error message as literal, this would be how I would write it:

```suggestion
    condition     = contains(["Basic", "VpnGw1", "VpnGw1AZ"], try(var.virtual_network_gateway_config.sku,[])) ? var.virtual_network_gateway_config.generation == "Generation1" : true
```

Meaning: 
If `var.virtual_network_gateway_config.sku` is one of `["Basic", "VpnGw1", "VpnGw1AZ"]`, then check if `var.virtual_network_gateway_config.generation` == `"Generation1"`, otherwise return true because this validation doesn't scope for other skus.

And then following this, you can add more validation blocks if you need to do more validation for other errors.

---

discussed and yes, suggestion makes sense.  original did not

removed brackets from other validation conditions as they're not needed